### PR TITLE
Add a new mixer API to fetch ALL landing page data and prototyped wit…

### PR DESCRIPTION
**Pull request by shifucun**
The data includes all the related places and the stats data for these places based on the chart config.

### CODE REVIEW

File: server/landing_page.go [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+	for _, n := range relatedPlaceData["Count_Person"].RelatedPlaces {
+		result = append(result, &place{
+			Dcid: n,
+		})
+	}
+	return result, nil
+}
+
+// GetLandingPageData implements API for Mixer.GetLandingPageData.
+// TODO(shifucun): implement similar and nearby places.
```
Comment (beets) [inline server/landing_page.go]: can remove this todo

File: server/landing_page.go [hunk 2]
```diff
@@ -413,8 +414,41 @@ func getSimilarPlaces(ctx context.Context, s *Server, dcid string) (
 	return result, nil
 }
 
+// Get nearby places.
```
Comment (beets) [inline server/landing_page.go:417]: the test response seems to be missing place names for nearby places

**Review by beets — COMMENTED** [review]
looks great. very nice to see this in go, thanks for migrating everything. mostly small comments.

File: server/landing_page.go [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+)
+
+type relatedPlace struct {
+	category string
+	places   []*place
+}
+
+var wantedPlaceTypes = map[string]map[string]struct{}{
+	"Country": {
+		"State":               {},
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/landing_page.go:58]: Go does not have a set and it's mostly implemented with map to struct: https://yourbasic.org/golang/implement-set/. This does look verbose..

File: server/landing_page.go [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+		for statVarDcid, obsTimeSeries := range landingPageData.Data {
+			obsTimeSeries.filterAndRank(&obsProp{})
+			finalData[statVarDcid] = obsTimeSeries
+		}
+		result[dcid] = finalData
+	}
+	return result, nil
+}
+
+// Pick child places with the most average population.
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/landing_page.go]: Done; Added a todo to pick secondary type if the set with highest population has too few children

File: server/landing_page.go [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+		}
+	}
+	if len(result) > maxNumChild {
+		result = result[0:maxNumChild]
+	}
+	return result
+}
+
+// Get child places by types.
+// The city under each type is sorted by the population.
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/landing_page.go]: Done

File: server/landing_page.go [hunk 6]
```diff
@@ -0,0 +1,498 @@
...
+	}
+	for _, children := range result {
+		for _, child := range children {
+			if val, ok := placePop[child.Dcid]; ok {
+				child.Pop = val
+			}
+		}
+	}
+	// Drop empty categories and sort the children by population
+	for typ := range result {
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/landing_page.go:294]: type is reserved in go, so "typ", although looks odd, is used commonly (can see a lot in cs)..

File: server/landing_page.go (hunk 1 above)
Comment (shifucun) [reply server/landing_page.go]: Done

File: server/landing_page.go (hunk 2 above)
Comment (shifucun) [reply server/landing_page.go:419]: Right. The response data is not consistent now. Some of them does not have names as the client now does not need them. Added a todo to get all name information here as well so client does not bother to fetch the name and even parent abbreviations.

**Review by beets — APPROVED** [review]

**Pull request merged**